tools-local/unix/tool-helper.sh

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/env bash
+
+# Provides helper functions.
+
+# Checks if the specified argument is an empty string.
+# If yes then, displays a message stating that the argument is empty, and exits with status 1.
+exit_if_arg_empty()
```

Comment:
Calling `using` would work here, but it wouldn't make sense when this function is used in the helper functions here. Maybe `$3` could be an optional name of a function to call on failure after echoing the error message, so the scripts can pass `using` and the helpers can pass nothing.